dev/deps/spark-deps-hadoop-2.7-hive-2.3

```diff
@@ -50,22 +46,20 @@ commons-lang3/3.12.0//commons-lang3-3.12.0.jar
 commons-logging/1.1.3//commons-logging-1.1.3.jar
 commons-math3/3.4.1//commons-math3-3.4.1.jar
 commons-net/3.1//commons-net-3.1.jar
-commons-pool/1.5.4//commons-pool-1.5.4.jar
 commons-text/1.6//commons-text-1.6.jar
 compress-lzf/1.0.3//compress-lzf-1.0.3.jar
 core/1.1.2//core-1.1.2.jar
 curator-client/2.7.1//curator-client-2.7.1.jar
 curator-framework/2.7.1//curator-framework-2.7.1.jar
 curator-recipes/2.7.1//curator-recipes-2.7.1.jar
-datanucleus-api-jdo/4.2.4//datanucleus-api-jdo-4.2.4.jar
 datanucleus-core/4.1.17//datanucleus-core-4.1.17.jar
-datanucleus-rdbms/4.1.19//datanucleus-rdbms-4.1.19.jar
 derby/10.14.2.0//derby-10.14.2.0.jar
-dropwizard-metrics-hadoop-metrics2-reporter/0.1.2//dropwizard-metrics-hadoop-metrics2-reporter-0.1.2.jar
+error_prone_annotations/2.5.1//error_prone_annotations-2.5.1.jar
+failureaccess/1.0.1//failureaccess-1.0.1.jar
 flatbuffers-java/1.9.0//flatbuffers-java-1.9.0.jar
 generex/1.0.2//generex-1.0.2.jar
 gson/2.2.4//gson-2.2.4.jar
-guava/14.0.1//guava-14.0.1.jar
+guava/30.1.1-jre//guava-30.1.1-jre.jar
```

Comment:
Since we use shaded guava now in both hive, hadoop, spark, can we remove this dep entirely?

Reply:
It actually might be possible: seems Spark now shade Guava usage everywhere within itself now. I remember seeing a few modules which are not using shaded version but now can't find them anymore. Perhaps some mistake from my side.

It's a bit strange that Spark packs the shaded Guava classes in its `spark-network-common` jar. This is working since it is dependent by `spark-core` and thus transitively dependent by other modules.

I need to fix the PR since the "inherited" tag will prevent Spark from shading jetty & guava in other modules, which is bad. Thinking about how to do proper plugin inheritance in Maven...

Reply:
Curator has a hard dependency on Guava, see detail at https://cwiki.apache.org/confluence/display/CURATOR/TN13

Reply:
Hmm you are right, I missed this. Somehow `mvn dependency:tree` doesn't show curator's guava dependency.

Reply:
So `build/sbt dependencyTree` provides a better result, and it looks like only `curator-client` and `hive-exec` has hard dependency on Guava at the moment. With the `hive-shaded` module, the latter is resolved so we are only left with the former, which makes me thinking whether we can just skip relocating the following 3 classes required by curator:

- com.google.common.base.Function
- com.google.common.base.Predicate
- com.google.common.reflect.TypeToken

It's not ideal but is better than exposing all the Guava classes.

Reply:
https://guava.dev/

> APIs without @Beta will remain binary-compatible for the indefinite future. (Previously, we sometimes removed such APIs after a deprecation period. The last release to remove non-@Beta APIs was Guava 21.0.) Even @Deprecated APIs will remain (again, unless they are @Beta). We have no plans to start removing things again, but officially, we’re leaving our options open in case of surprises (like, say, a serious security problem).

I think it's fine to 1)use vanilla Guava jar directly, 2) avoid using Guava @Beta APIs in Spark, 3)keep Guava in the latest version.

Reply:
Thanks for the reference! this is good from Spark side but it won't help if downstream apps use beta API which were changed and thus causing conflicts. I agree with you that we should avoid using these beta APIs - perhaps worth adding a maven plugin to check this.

In the mean time I'll leave the guava dependency in the assembly module and we should revisit it separately. Ideally I feel we should remove it if only `curator-client` is blocking it but we'll need to double check on that.